Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp

```diff
@@ -392,20 +396,42 @@ static void bindGStreamerData(Vector<CString>& args)
 
     if (!bindPathVar(args, "GST_PLUGIN_SYSTEM_PATH_1_0")) {
         if (!bindPathVar(args, "GST_PLUGIN_SYSTEM_PATH")) {
+            // This looks dodgy, why would system plugins be stored in the user data dir?
             GUniquePtr<char> gstData(g_build_filename(g_get_user_data_dir(), "gstreamer-1.0", nullptr));
             bindIfExists(args, gstData.get());
         }
     }
 
-    GUniquePtr<char> gstCache(g_build_filename(g_get_user_cache_dir(), "gstreamer-1.0", nullptr));
-    bindIfExists(args, gstCache.get(), BindFlags::ReadWrite);
+    // The plugin scanner needs write permissions in the parent directory of GST_REGISTRY in order to
+    // write the registry file.
+    GUniquePtr<char> defaultRegistryPath(g_build_filename(g_get_user_cache_dir(), "gstreamer-1.0", nullptr));
```

Comment:
Hmmm, well if write permissions are truly required, then it's required, but: if you give write permission here, then host GStreamer MUST be robust to malicious files in its cache. Hopefully this is actually true? Otherwise, it's a sandbox escape. The whole sandbox is only as good as the weakest link.

Maybe it would make sense to use a separate registry cache somewhere where the host will not accidentally use it, and have bwrap set the GST_REGISTRY environment variable to point to it? Then the write permissions are much less risky, right?

(In contrast, the write permissions for debug environment variables are not so scary because you know that by setting the environment variable, you definitely want GStreamer to write to that location.)

Reply:
You mean, if GST_REGISTRY is set, we would re-set it to a different "safe" hardcoded path?

Reply:
Hm, no. If it's set, then you really want to use a particular registry path, and we should respect that. That's an explicit request by the user to allow the web process to use the registry.

I was thinking "if it's unset, then set it to someplace safe where the host will not see it," but thinking this through more, we could accomplish the same by just not mounting the default registry path, right? And that's already the current behavior. Do you agree? We should mount the registry path only if the environment variable is set?

Reply:
OK, I think this is addressed in the latest version. Thanks for the review! I'll land this in a couple days, giving time to @TingPing to also give feedback.

Reply:
As long as these are intended for developers it sounds fine to me.